pkg/dao/rpm.go

```diff
@@ -0,0 +1,200 @@
+package dao
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/content-services/content-sources-backend/pkg/api"
+	"github.com/content-services/content-sources-backend/pkg/models"
+	"gorm.io/gorm"
+)
+
+type rpmDaoImpl struct {
+	db *gorm.DB
+}
+
+func GetRpmDao(db *gorm.DB) RpmDao {
+	return rpmDaoImpl{
+		db: db,
+	}
+}
+
+func (r rpmDaoImpl) isOwnedRepository(orgID string, repoUUID string) (bool, error) {
+	var repoConfigs []models.RepositoryConfiguration
+	var count int64
+	if err := r.db.
+		Where("org_id = ? and repository_uuid = ?", orgID, repoUUID).
+		Find(&repoConfigs).
+		Count(&count).
+		Error; err != nil {
+		return false, err
+	}
+	if count == 0 {
+		return false, nil
+	}
+	return true, nil
+}
+
+// Create a record in rpms table, and the relation between it
+// and the Repository record in the repositories_rpms table.
+// Before create the record, it checks that the provided
+// Repository register is related with a RepositoryConfiguration
+// that belong to the indicated organization.
+// orgID It is used to check the repository which we are trying
+// to add the Rpm record belongs to the indicated organization.
+// repo The repository record that the new rpm record will
+// belong to.
+// newRpm The Rpm record to be created into the database.
+// Return error if something goes wrong, else nil.
+func (r rpmDaoImpl) Create(orgID string, repo *models.Repository, newRpm *models.Rpm) error {
```

Comment:
also normally we'd not have Create() accept a model, it would accept an api struct, but since this doesn't make sense for Rpm (you wouldn't create them from the api), i think making it private  keeps the spirit of the separation.

Reply:
thanks! I have deleted the no required methods from the interface and updated the tests